vector/src/main/java/im/vector/app/features/MainActivity.kt

```diff
@@ -147,38 +150,41 @@ class MainActivity : VectorBaseActivity<FragmentLoadingBinding>(), UnlockedActiv
         }
         when {
             args.isAccountDeactivated -> {
-                // Just do the local cleanup
-                Timber.w("Account deactivated, start app")
-                sessionHolder.clearActiveSession()
-                doLocalCleanup(clearPreferences = true)
-                startNextActivityAndFinish()
+                emptyViewModel.viewModelScope.launch {
```

Comment:
Why not dispatching to a real viewmodel?

Reply:
This is another rework... But yes, ideally it will have to be done